src/System.Linq.Parallel/src/System/Linq/Parallel/Channels/SynchronousChannel.cs

```diff
@@ -29,15 +29,7 @@ internal sealed class SynchronousChannel<T>
 #if DEBUG
         // In debug builds, we keep track of when the producer is done (for asserts).
         private bool _done;
-#endif
-
-        //-----------------------------------------------------------------------------------
-        // Instantiates a new queue.
-        //
-
-        internal SynchronousChannel()
```

Comment:
It may never be a problem but one possible future issue could be we decide to make this type public and it accidently now has a public constructor when we didn't realize it where as having the explicit internal constructor forces us to make the decision about making that public or not.